vscode-extension/src/compiler.ts

```diff
@@ -6,52 +6,50 @@
  */
 
 import {window} from 'vscode';
-import {RelayExtensionContext} from './context';
+import {RelayProjectExtensionContext} from './context';
 import {getConfig} from './config';
 
-export function createAndStartCompiler(context: RelayExtensionContext) {
-  if (context.compilerTerminal) {
+export function createAndStartCompiler(context: RelayProjectExtensionContext) {
+  if (context.project.compilerTerminal) {
     return;
   }
 
   const config = getConfig();
 
-  const args: string[] = ['--watch', `--output=${config.compilerOutpuLevel}`];
+  const args: string[] = ['--watch', `--output=${config.compilerOutputLevel}`];
 
-  if (config.pathToConfig) {
-    args.push(config.pathToConfig);
+  if (context.project.binaryExecutionOptions.pathToConfig) {
+    args.push(context.project.binaryExecutionOptions.pathToConfig);
   }
 
   const terminal = window.createTerminal({
     name: 'Relay Compiler',
-    cwd: context.relayBinaryExecutionOptions.rootPath,
+    cwd: context.project.binaryExecutionOptions.rootPath,
   });
 
   terminal.sendText(
-    `${context.relayBinaryExecutionOptions.binaryPath} ${args.join(' ')}`,
+    `${context.project.binaryExecutionOptions.binaryPath} ${args.join(' ')}`,
   );
 
-  terminal.show();
```

Comment:
This disables the terminal auto-showing when switching to a workspace that starts the compiler by default. I don't see much reason to auto-show this when switching workspaces and personally find it a bit annoying